pallets/schemas/src/lib.rs

```diff
@@ -209,6 +233,21 @@ pub mod pallet {
 		OptionQuery,
 	>;
 
+	/// Storage for message schema info struct data
+	/// - Key: Schema Id
+	/// - Value: [`SchemaInfo`](SchemaInfo)
+	#[pallet::storage]
+	#[pallet::getter(fn get_schema_id)]
+	pub(super) type SchemaNamesToId<T: Config> = StorageDoubleMap<
```

Comment:
**Question:** `StorageDoubleMa` or `StorageNMap`?

name format: `<namespace>.<descriptor>`
`StorageNMap` will allow us to do prefix queries on namespaces. but since we are hashing on namespace and descriptor separately it might be easier to create unbalanced trie but on the other hand since these are going through governanace maybe that is not an issue.

open to suggestions

Reply:
Hmm... I like the idea of `nmap`. Clarifies the requirements around each piece and the name becomes more `namespace.descriptor.version` One can get any superset of that.